bahmni_sale/models/sale_order_line.py

```diff
@@ -23,6 +23,8 @@ class SaleOrderLine(models.Model):
     def onchange_lot_id(self):
         if self.lot_id:
             self.expiry_date = self.lot_id.life_date
+            if self.env.ref('bahmni_sale.sale_price_basedon_cost_price_markup').value == '1':
+                self.price_unit = self.lot_id.sale_price if self.prod_lot.sale_price > 0.0 else self.price_unit
```

Comment:
Same here. is the "else" statement required? we are just replacing with the same value from self.price_unit.

Reply:
Actually if we are assigning the sale price of the lot_id but if the price is 0.0 means not set then we cant get that value so we will make sure that we will not apply it. So whatever the price of the previously set by product will remain same.